test/server_test.cpp

```diff
@@ -0,0 +1,35 @@
+#include <sstream>
+#include <string>
+#include <iostream>
+
+#include "gtest/gtest.h"
+#include "server.hpp"
+#include <signal.h>
+
+
+TEST(ServerTest, PortTooBig) {
+	std::string addr = "0.0.0.0";
+	std::string port = "12312312";
+
+	http::server::server server_(addr, port);
+	EXPECT_FALSE(server_.isValid(addr, port));
+}
+
+TEST(ServerTest, PortIsNegative) {
```

Comment:
Oh okay, let's leave the port validation code in main then -- it'll catch all of the invalid port stuff before it even reaches server.cpp. And then we can have the tests relating to out-of-range port numbers in main_test.cpp